internal/connector/pingone_platform/resources/pingone_branding_settings.go

```diff
@@ -0,0 +1,62 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/pingidentity/pingctl/internal/connector"
+	"github.com/pingidentity/pingctl/internal/logger"
+)
+
+// Verify that the resource satisfies the exportable resource interface
+var (
+	_ connector.ExportableResource = &PingoneBrandingSettingsResource{}
+)
+
+type PingoneBrandingSettingsResource struct {
+	clientInfo *connector.SDKClientInfo
+}
+
+// Utility method for creating a PingoneAgreementLocalizationRevisionResource
+func BrandingSettingsResource(clientInfo *connector.SDKClientInfo) *PingoneBrandingSettingsResource {
+	return &PingoneBrandingSettingsResource{
+		clientInfo: clientInfo,
+	}
+}
+
+func (r *PingoneBrandingSettingsResource) ExportAll() (*[]connector.ImportBlock, error) {
+	l := logger.Get()
+
+	l.Debug().Msgf("Fetching all pingone_branding_settings resources...")
+
+	brandingSettings, response, err := r.clientInfo.ApiClient.ManagementAPIClient.BrandingSettingsApi.ReadBrandingSettings(r.clientInfo.Context, r.clientInfo.EnvironmentID).Execute()
+	defer response.Body.Close()
+	if err != nil {
+		l.Error().Err(err).Msgf("ReadBrandingSettings Response Code: %s\nResponse Body: %s", response.Status, response.Body)
+		return nil, err
+	}
+
+	if brandingSettings == nil {
+		l.Error().Msgf("Returned ReadBrandingSettings() settings are nil.")
+		l.Error().Msgf("ReadBrandingSettings Response Code: %s\nResponse Body: %s", response.Status, response.Body)
+		return nil, fmt.Errorf("failed to fetch pingone_branding_settings resources via ReadBrandingSettings()")
+	}
+
+	importBlocks := []connector.ImportBlock{}
+
+	// Oddly, environment is nil in the returned brandingSettigs, but id seems to be set to the env ID
+	environmentId, environmentIdOk := brandingSettings.GetIdOk()
+
+	if environmentIdOk {
+		importBlocks = append(importBlocks, connector.ImportBlock{
```

Comment:
With singleton resources like this (I believe most "settings" resources are singletons) there shouldn't be any need to make any API call. The only thing needed is the environment ID, which we already have from the clientInfo. With import blocks, we only need requests when there's other elements needed for the provider to perform a terraform import.

We may want to have some validation that the provided environment ID exists at all but I think we'd want to do that at the connector level rather than the resource level.

Reply:
That makes sense, I've added some logic to validate environment IDs on the connector level, as well as added the user ability to specify an env id to export separately from the worker env id.

Updated each resource to use the clientInfo env id directly.

Finally, updated region codes to use full strings to match pingone provider and expected values for the SDK client config.